lib/common.bash

```diff
@@ -532,3 +532,354 @@ check_dockerfiles_images()
 		build_dockerfile_image "$image" "$dockerfile_path"
 	fi
 }
+
+###############################################################################
+
+# Misc functions.
+
+TESTS_REPO_DIR=$(realpath $(dirname "${BASH_SOURCE[0]}")/..)
+
+# sudo shortcut function with environment
+esudo() {
+  sudo -E PATH=$PATH "${@}"
+}
+
+# Get version of component from tests versions.yaml
+tests_get_version() {
+  local dependency="${1}"
+  local versions_file="${TESTS_REPO_DIR}/versions.yaml"
+
+  # Error if versions file not present
+  [ -f "${versions_file}" ] || (>&2 echo "Cannot find ${versions_file}"; return 1)
+
+  # Parse versions file with yq for dependency
+  result=$("${GOPATH}/bin/yq" r -X "${versions_file}" "${dependency}")
+  [ "${result}" = "null" ] && result=""
+  echo "${result}"
+}
+
+# SSH grep for a string in the target dmesg output
+ssh_dmesg_grep() {
+  local private_key_file="${1}"
+  local pod_ip="${2}"
+  local grep_text="${3}"
+
+  ssh -i ${private_key_file} \
+    -o "StrictHostKeyChecking no" \
+    -o "PasswordAuthentication=no" \
+    -o ConnectTimeout=1 \
+    -t root@${pod_ip} \
+    "dmesg 2>&1 | grep \"${grep_text}\"" || true
+}
+
+
+###############################################################################
+
+# docker Image Handling
+
+# Retrieve encryption key from docker image label
+# 'enc_key' image label must be set when image created or uploaded
+docker_image_label_get_encryption_key() {
+  local image_url="${1}"
+
+  # Pull image
+  esudo docker pull "${image_url}" &>/dev/null
+
+  # Get encryption key from docker image label
+  esudo docker inspect ${image_url} \
+    | jq -r '.[0].Config.Labels.enc_key'
+}
+
+# Retrieve ssh key from docker image label
+# 'ssh_key' image label must be set when image created or uploaded
+docker_image_label_save_ssh_key() {
+  local image_url="${1}"
+  local ssh_key_file="${2}"
+
+  # Pull image
+  esudo docker pull "${image_url}"
+
+  # Get ssh key from docker image label and save to file
+  esudo docker inspect ${image_url} \
+    | jq -r '.[0].Config.Labels.ssh_key' \
+    | sed "s|\(-----BEGIN OPENSSH PRIVATE KEY-----\)|\1\n|g" \
+    | sed "s|\(-----END OPENSSH PRIVATE KEY-----\)|\n\1|g" \
+    > "${ssh_key_file}"
+
+  # Set permissions on private key file
+  chmod 600 "${ssh_key_file}"
+}
+
+
+###############################################################################
+
+# k8s
+
+TESTS_REPO_DIR=$(realpath $(dirname "${BASH_SOURCE[0]}")/..)
+FIXTURES_DIR="${TESTS_REPO_DIR}/integration/kubernetes/confidential/fixtures"
+
+# Generate kubernetes service yaml from template
+k8s_generate_service_yaml() {
```

Comment:
Ah, also, If you look at `integration/kubernetes/confidential/lib.sh` you will see the function's names start with `kubernetes_` instead of `k8s_`. So I suggest to rename these new functions.

Reply:
Renamed.